codalab/worker/worker_run_state.py

```diff
@@ -645,10 +699,17 @@ def _transition_from_FINALIZING(self, run_state):
         move on to the FINISHED state. Can also remove bundle_path now.
         """
         if run_state.is_restaged:
+            logger.info(f'{run_state.bundle.uuid} is restaged')
+            log_bundle_transition(
+                run_state.bundle.uuid, run_state.stage, RunStage.RESTAGED, 'it has finalized'
+            )
             return run_state._replace(stage=RunStage.RESTAGED)
         elif run_state.finalized:
             if not self.shared_file_system:
                 remove_path(run_state.bundle_path)  # don't remove bundle if shared FS
+            log_bundle_transition(
+                run_state.bundle.uuid, run_state.stage, RunStage.FINISHED, 'it has finalized'
```

Comment:
Stating that the bundle transitioned to FINISHED is enough. I don't think we need the reason here.